core/src/main/scala/kafka/admin/ConsumerGroupCommand.scala

```diff
@@ -283,50 +363,55 @@ object ConsumerGroupCommand {
           .orElse(throw new BrokerNotAvailableException("Broker id %d does not exist".format(brokerId)))
       } catch {
         case t: Throwable =>
-          println("Could not parse broker info due to " + t.getMessage)
+          consumerGroupOutputWriter.printError("Could not parse broker info due to " + t.getMessage, Some(t))
           None
       }
     }
 
   }
 
-  class KafkaConsumerGroupService(val opts: ConsumerGroupCommandOptions) extends ConsumerGroupService {
+  class KafkaConsumerGroupService(val opts: ConsumerGroupCommandOptions, val consumerGroupOutputWriter: ConsumerGroupOutputWriter) extends ConsumerGroupService {
 
     private val adminClient = createAdminClient()
 
     // `consumer` is only needed for `describe`, so we instantiate it lazily
     private var consumer: KafkaConsumer[String, String] = null
 
-    def list() {
-      adminClient.listAllConsumerGroupsFlattened().foreach(x => println(x.groupId))
+    def listGroups(): List[String] = {
+      adminClient.listAllConsumerGroupsFlattened().map { _.groupId }.toList
     }
 
-    protected def describeGroup(group: String) {
+    protected def collectGroupAssignment(group: String): Option[Array[PartitionAssignmentState]] = {
       adminClient.describeConsumerGroup(group) match {
-        case None => println(s"Consumer group `${group}` does not exist.")
+        case None =>
+          consumerGroupOutputWriter.printError(s"Consumer group '$group' does not exist.")
```

Comment:
Looks like we would print this twice. Once here and once in main. I'm wondering if we could remove `ConsumerGroupOutputWriter` from this class and do all the output in main. Maybe we just need to move the empty check that you have below in main?